core/src/main/scala/org/apache/spark/metrics/ExecutorMetricType.scala

```diff
@@ -99,6 +100,61 @@ case object ProcessTreeMetrics extends ExecutorMetricType {
   }
 }
 
+case object GarbageCollectionMetrics extends ExecutorMetricType {
+  import GC_TYPE._
+  override val names = Seq(
+    "MinorGCCount",
+    "MinorGCTime",
+    "MajorGCCount",
+    "MajorGCTime"
+  )
+  override private[spark] def getMetricValues(memoryManager: MemoryManager): Array[Long] = {
+    var allMetrics = GCMetrics(0, 0, 0, 0)
```

Comment:
IMHO `GCMetrics` doesn't really make things any simpler over just creating an `Array[Long](4)` and updating it directly since its tightly scoped.

Reply:
Agree